### PR TITLE
Add AccessRequest access to userACL

### PR DESCRIPTION
**Pull request by kimlisa**
part of https://github.com/gravitational/teleport/issues/4937

#### Description
- Add `AccessRequest` perms to userACL
- Get/Set requestable roles from oracle

### CODE REVIEW

**Review by alex-kovoy — APPROVED** [review]

File: lib/web/ui/usercontext_test.go [hunk 1]
```diff
@@ -71,6 +71,7 @@ func (s *UserContextSuite) TestNewUserContext(c *check.C) {
 	c.Assert(userContext.ACL.Users, check.DeepEquals, denied)
 	c.Assert(userContext.ACL.Tokens, check.DeepEquals, denied)
 	c.Assert(userContext.ACL.Nodes, check.DeepEquals, denied)
+	c.Assert(userContext.ACL.AccessRequest, check.DeepEquals, denied)
```
Comment (alex-kovoy) [inline lib/web/ui/usercontext_test.go:74]: `userContext.ACL.AccessRequest` -> `AccessRequests`

Comment (kimlisa) [reply lib/web/ui/usercontext_test.go]: revised

**Review by alex-kovoy — APPROVED** [review]

**Review by alex-kovoy — APPROVED** [review]

**Review by russjones — APPROVED** [review]

File: docs/testplan.md [hunk 2]
```diff
@@ -271,6 +271,84 @@ For main, test with admin role that has access to all resources.
 - [ ] Verify event detail dialogue renders when clicking on events `details` button
 - [ ] Verify searching by type, description, created works
 
+#### Access Requests
+1. Create a role with limited permissions (defined below as `allow-roles`). This role allows you to see Role screen, and allow you to ssh into all nodes.
```
Comment (a-palchikov) [inline docs/testplan.md:275]:
```suggestion
1. Create a role with limited permissions (defined below as `allow-roles`). This role allows you to see the Role screen and ssh into all nodes.
```

File: docs/testplan.md [hunk 3]
```diff
@@ -271,6 +271,84 @@ For main, test with admin role that has access to all resources.
...
+      - list
+      - read
+      - create
+  options:
+    max_session_ttl: 8h0m0s
+version: v3
+```
+- [ ] Verify that requests are shown and that correct states are applied to each request (pending, approved, denied)
+- [ ] Verify that creating a new request works
+  - [ ] Verify that under requestable roles, only `allow-roles` and `allow-users` is listed
```
Comment (a-palchikov) [inline docs/testplan.md:343]:
```suggestion
  - [ ] Verify that under requestable roles, only `allow-roles` and `allow-users` are listed
```

File: docs/testplan.md [hunk 4]
```diff
@@ -271,6 +271,84 @@ For main, test with admin role that has access to all resources.
...
+  - [ ] Verify input validation requires at least one role to be selected
+- [ ] Verify assume buttons are only present for approved request and for logged in user
+  - [ ] Verify assuming `allow-roles` allows you to see roles screen and ssh into nodes
```
Comment (a-palchikov) [inline docs/testplan.md:346]:
```suggestion
  - [ ] Verify that assuming `allow-roles` allows you to see roles screen and ssh into nodes
```

File: docs/testplan.md [hunk 5]
```diff
@@ -271,6 +271,84 @@ For main, test with admin role that has access to all resources.
...
+  - [ ] Verify after clicking on assume buttons, button is disabled
```
Comment (a-palchikov) [inline docs/testplan.md:347]:
```suggestion
  - [ ] Verify that after clicking on the assume button, it is disabled
```

File: docs/testplan.md [hunk 6]
```diff
@@ -271,6 +271,84 @@ For main, test with admin role that has access to all resources.
...
+  - [ ] After assuming `allow-roles`, verify assuming `allow-users` allows you to see users screen, and denies access to nodes.
```
Comment (a-palchikov) [inline docs/testplan.md:348]:
```suggestion
  - [ ] After assuming `allow-roles`, verify that assuming `allow-users` allows you to see users screen, and denies access to nodes.
```

File: docs/testplan.md [hunk 7]
```diff
@@ -271,6 +271,84 @@ For main, test with admin role that has access to all resources.
...
+    - [ ] Verify after 4 minutes, user is logged out of screen automatically.
```
Comment (a-palchikov) [inline docs/testplan.md:349]:
```suggestion
    - [ ] Verify that after 4 minutes, the user is automatically logged out.
```

File: docs/testplan.md [hunk 8]
```diff
@@ -271,6 +271,84 @@ For main, test with admin role that has access to all resources.
...
+- [ ] Verify after logging out (or getting logged out automatically) and relogging in, permissions are set back to `default`, and requests that are not expired and are approved, is assumable again.
```
Comment (a-palchikov) [inline docs/testplan.md:350]:
```suggestion
- [ ] Verify that after logging out (or getting logged out automatically) and relogging in, permissions are reset to `default`, and requests that are not expired and are approved are assumable again.
```

File: docs/testplan.md [hunk 9]
```diff
@@ -494,7 +558,17 @@ Add the following resource to enable read access to trusted clusters
 - [ ] Verify that a user can access the "Trust" screen
 - [ ] Verify that a user cannot create/delete/update a trusted cluster.
 
+Add the following resource to enable read access to access_request
```
Comment (a-palchikov) [inline docs/testplan.md:561]:
```suggestion
Add the following to enable read access to the access_request resource
```

File: docs/testplan.md [hunk 10]
```diff
@@ -494,7 +558,17 @@ Add the following resource to enable read access to trusted clusters
...
 
+```
+- resources:
+      - access_request
+      verbs:
+      - list
+      - read
+```
+- [ ] Verify that a user can see the "Access Request" screen
+* Note: users are always allowed to create their own request, if they have any requestable roles
```
Comment (a-palchikov) [inline docs/testplan.md:571]:
```suggestion
* Note: users are always allowed to create their own requests, if they have any requestable roles
```

**Review by a-palchikov — APPROVED** [review]

File: docs/testplan.md (hunk 10 above)
Comment (kimlisa) [reply docs/testplan.md:571]: revised

File: docs/testplan.md (hunk 9 above)
Comment (kimlisa) [reply docs/testplan.md:561]: revised

File: docs/testplan.md (hunk 8 above)
Comment (kimlisa) [reply docs/testplan.md:350]: revised

File: docs/testplan.md (hunk 7 above)
Comment (kimlisa) [reply docs/testplan.md:349]: revised

File: docs/testplan.md (hunk 6 above)
Comment (kimlisa) [reply docs/testplan.md:348]: revised

File: docs/testplan.md (hunk 5 above)
Comment (kimlisa) [reply docs/testplan.md:347]: revised

File: docs/testplan.md (hunk 4 above)
Comment (kimlisa) [reply docs/testplan.md:346]: revised

File: docs/testplan.md (hunk 3 above)
Comment (kimlisa) [reply docs/testplan.md:343]: revised

File: docs/testplan.md (hunk 2 above)
Comment (kimlisa) [reply docs/testplan.md:275]: revised

**Pull request merged**